impl/asm-based-transformer/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -226,6 +228,86 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
+                         * ConstantDynamic can modify content of array at runtime.
+                         */
+                        Object[] copyBootstrapMethodArguments = null;
+                        for(int looper = 0; looper < bootstrapMethodArguments.length; looper++) {
+                            Object argument = bootstrapMethodArguments[looper];
+                            if (argument instanceof Type) {
+                                Type type = (Type) argument;
+                                String oldDesc = type.getDescriptor();
+                                String updatedDesc = replaceJavaXwithJakarta(type.getDescriptor());
+                                if (!oldDesc.equals(updatedDesc)) {
+                                    setClassTransformed(true);
+                                    if (copyBootstrapMethodArguments == null) {
+                                        copyBootstrapMethodArguments = cloneBootstrapMethodArguments(bootstrapMethodArguments);
+                                    }
+                                    copyBootstrapMethodArguments[looper] = Type.getMethodType(updatedDesc);
+                                }
+                            } else if (argument instanceof Handle) {  // reference to a field or method
+                                Handle handle = (Handle)argument;
+                                String origDesc = handle.getDesc();
+                                String updatedDesc = replaceJavaXwithJakarta(handle.getDesc());
+                                if (!origDesc.equals(updatedDesc)) {  // if we are changing
+                                    // mark the class as transformed
+                                    setClassTransformed(true);
+                                    handle = new Handle(
+                                            handle.getTag(),
+                                            handle.getOwner(),
+                                            handle.getName(),
+                                            updatedDesc,
+                                            handle.isInterface());
+                                    if (copyBootstrapMethodArguments == null) {
+                                        copyBootstrapMethodArguments = cloneBootstrapMethodArguments(bootstrapMethodArguments);
+                                    }
+                                    copyBootstrapMethodArguments[looper] = handle;
+                                }
+                                
+                            } else if( argument instanceof ConstantDynamic) {
+                                // TODO: runtime handling of ConstantDynamic)
+                                ConstantDynamic constantDynamic = (ConstantDynamic)argument;
+                                throw new IllegalStateException("ConstantDynamic is not handled " +constantDynamic.toString());
```

Comment:
I see https://github.com/wildfly-extras/batavia/issues/28 was created to cover that.